src/mlpack/core/data/tokenizers/bert_tokenizer_impl.hpp

```diff
@@ -0,0 +1,276 @@
+/**
+ * @file core/data/tokenizers/bert_tokenizer_impl.hpp
+ * @author Ayush Singh
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_TOKENIZERS_BERT_TOKENIZER_IMPL_HPP
+#define MLPACK_CORE_DATA_TOKENIZERS_BERT_TOKENIZER_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "bert_tokenizer.hpp"
+
+namespace mlpack {
+namespace data {
+
+static bool isStripChar(const wchar_t& ch) {
+    return stripChar.find(ch) != std::wstring::npos;
+}
+
+static std::wstring strip(const std::wstring& text) {
+    std::wstring ret =  text;
+    if (ret.empty()) return ret;
+    size_t pos = 0;
+    while (pos < ret.size() && isStripChar(ret[pos])) pos++;
+    if (pos != 0) ret = ret.substr(pos, ret.size() - pos);
+    pos = ret.size() - 1;
+    while (pos != (size_t)-1 && isStripChar(ret[pos])) pos--;
+    return ret.substr(0, pos + 1);
+}
+
+static std::vector<std::wstring> split(const std::wstring& text) {
+    std::vector<std::wstring>  result;
+    boost::split(result, text, boost::is_any_of(stripChar));
```

Comment:
Would you mind explaining me the difference between `BasicTokenizer` , We aldeardy have a Basic Tokenizer implemented in mlpack 

https://github.com/mlpack/mlpack/blob/28d03e5dd73245e29212caae31a75b9874661eb0/src/mlpack/core/data/tokenizers/split_by_any_of.hpp#L26

Reply:
You are using the `boost::is_any_of()` function to split , And that is the same we have implemented in mlpack without the boost